frontend/src/pages/Profile.js

```diff
@@ -1,58 +1,188 @@
     this.setTitle('Profile');
   }
 
-  async getUserInfo() {
-    const userProfileInfo = await fetch('http://localhost:8080/userInfo').then(
-      (res) => res.json()
-    );
-    return userProfileInfo;
+  async getProfile() {
+    return Fetch.get('/users/profile/').catch(() => {
+      ErrorHandler.setToast('Failed to get user info');
+      return [];
+    });
   }
 
-  async render() {
-    const editModalBtn = openModalButton({
+  async getPageData() {
+    const profileData = await this.getProfile();
+
+    if (profileData.length === 0) {
+      return `
+        <div class="d-flex justify-content-center align-items-center h-100">
+          <div class="fs-11"> No User Profile :( </div>
+        </div>
+      `;
+    }
+
+    const profile = Object.keys(profileData)
+      .map((key) => {
+        if (key === 'image' || key === 'battleHistory' || key === 'is_active')
+          return '';
+        return `${ProfileItem({ type: key, content: profileData[key] ?? '' })}`;
+      })
+      .join('');
+
+    const editModalBtn = OpenModalButton({
       text: '>> EDIT PROFILE <<',
       classList: 'btn btn-no-outline-hover fs-8',
-      modalId: '#editProfile',
+      modalId: '#editProfileModal',
+    });
+    const editModal = ModalComponent({
+      borderColor: 'pink',
+      title: 'EDIT PROFILE',
+      modalId: 'editProfileModal',
+      content: EditProfileForm({
+        nick: profileData.username,
+        comment: profileData.comment ?? 'Please write a comment',
+      }),
+      buttonList: ['profileEditBtn'],
     });
-    const editModal = ({ nickname, comment }) =>
-      modalComponent({
-        borderColor: 'pink',
-        title: 'EDIT PROFILE',
-        modalId: 'editProfile',
-        content: editProfile({ nickname, comment }),
-        buttonList: ['confirmBtn'],
-      });
-
-    const dummyUser = await this.getUserInfo();
 
+    // TODO: battleHistory 정보 요청
+    // TODO: image url 정보 요청
     return `
-      <h1 class="fs-15">PLAYER PROFILE</h1>
-      <div class="container h-100 overflow-auto" style="width: 80%">
-        <div class="row d-flex justify-content-center">
-          ${UserProfile({
-            nickname: dummyUser.nickname,
-            email: dummyUser.email,
-            winLose: `${dummyUser.win} / ${dummyUser.lose}`,
-            comment: dummyUser.comment,
-            img: dummyUser.img,
-            battle: dummyUser.battleHistory,
-          })}
-        </div>
         <div class="row d-flex justify-content-center">
-          ${editModalBtn}
-          ${editModal({ nickname: dummyUser.nickname, comment: dummyUser.comment })}
+          <div class="row d-flex flex-row mt-4">
+            <div class="col-9 px-4 fs-7">
+              ${profile}
+            </div>
+            <div class="col-3 p-2 h-90">
+              <div class="h-100 w-100 d-flex justify-content-center align-items-center border border-2 border-light rounded">
+                <label for="profileImg" class="h-100 w-100">
```

Comment:
```suggestion
                <label for="profileImg" class="h-100 w-100 d-flex justify-content-center align-items-center">
```
사진 가운데 정렬 해주면 좋을 것 같아요!